src/codegen/generate.ts

```diff
@@ -45,13 +45,18 @@ export function getBodyFormatter(body?: OpenAPIV3.RequestBodyObject) {
   }
 }
 
+type SchemaMetadata = {
+  selfRef?: string;
+};
```

Comment:
Can we use a more single-purpose structure for this? 
`metaData.selfRef` Isn't really telling. 

How about `x-component-ref-path`?